canonicalwebteam/discourse_docs/parsers.py

```diff
@@ -45,32 +45,60 @@ def parse(self):
             features="html.parser",
         )
 
-        # Parse URL & redirects mappings (get warnings)
-        self.url_map, url_warnings = self._parse_url_map(raw_index_soup)
-        self.redirect_map, redirect_warnings = self._parse_redirect_map(
-            raw_index_soup
-        )
-        self.warnings = url_warnings + redirect_warnings
-
-        # Get body and navigation HTML
-        self.index_document = self.parse_topic(index_topic)
-        index_soup = BeautifulSoup(
-            self.index_document["body_html"], features="html.parser"
-        )
-        self.index_document["body_html"] = str(
-            self._get_preamble(index_soup, break_on_title="Navigation")
-        )
+        # Parse engage pages
+        if index_topic["id"] == 17229:
```

Comment:
I really don't think this is a good way to differentiate between parsing engage pages and parsing documents.

If there's significant differences in how we parse them (and I expect them to be) then I would think the clear way to write this module would be to actually make it a different class:

```
from canonicalwebteam.discourse import EngagePages

engage_pages = EngagePages(api_key=xxxx, ...)
```

Of course you would make `EngagePages` extend or otherwise share most of its code with the `DiscourseDocs` (or maybe simply `Docs`, but ubuntu.com certainly shouldn't be accessing engage pages through a class called `DiscourseDocs`, but just with a special index topic ID of 17229. That's hella confusing.